cmd/worker/internal/sourcegraphaccounts/notifications_subscriber.go

```diff
@@ -0,0 +1,235 @@
+package sourcegraphaccounts
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sourcegraph/log"
+	sams "github.com/sourcegraph/sourcegraph-accounts-sdk-go"
+	clientsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1"
+	notificationsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/notifications/v1"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+	"golang.org/x/oauth2/google"
+
+	"github.com/sourcegraph/sourcegraph/cmd/worker/job"
+	workerdb "github.com/sourcegraph/sourcegraph/cmd/worker/shared/init/db"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/dotcom"
+	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+var _ job.Job = (*notificationsSubscriber)(nil)
+
+// notificationsSubscriber is a worker responsible for receiving notifications
+// from Sourcegraph Accounts.
+type notificationsSubscriber struct {
+	config *notificationsSubscriberConfig
+}
+
+func NewNotificationsSubscriber() job.Job {
+	return &notificationsSubscriber{
+		config: &notificationsSubscriberConfig{},
+	}
+}
+
+func (s *notificationsSubscriber) Description() string {
+	return "Receives notifications from Sourcegraph Accounts."
+}
+
+func (s *notificationsSubscriber) Config() []env.Config {
+	return []env.Config{s.config}
+}
+
+func (s *notificationsSubscriber) Routines(ctx context.Context, observationCtx *observation.Context) ([]goroutine.BackgroundRoutine, error) {
+	logger := observationCtx.Logger.Scoped("sourcegraphAccounts.notificationsSubscriber")
+	if !dotcom.SourcegraphDotComMode() || s.config.GCP.CredentialsFile == "" {
+		logger.Debug("worker disabled because it is not running on Sourcegraph.com or missing GCP credentials file")
+		return nil, nil
+	}
+
+	// NOTE: Theoretically, we could have multiple SAMS providers configured, but in
+	// practice, we should ever only have one in production. Otherwise, we might
+	// have a bigger problem to address than just this worker.
+	var samsProvider *schema.OpenIDConnectAuthProvider
+	authProviders := conf.Get().SiteConfig().AuthProviders
+	for _, p := range authProviders {
+		if p.Openidconnect != nil && strings.HasPrefix(p.Openidconnect.ClientID, "sams_cid_") {
+			samsProvider = p.Openidconnect
+			break
+		}
+	}
+	if samsProvider == nil {
+		logger.Debug("worker disabled because SAMS provider is not configured")
```

Comment:
The default state in dotcom is this is enabled, so IMO disabled state should be more visible

```suggestion
		logger.Info("worker disabled because SAMS provider is not configured")
```